editor/src/components/canvas/dom-walker.ts

```diff
@@ -1045,9 +1038,58 @@ function getSpecialMeasurements(
     fontWeight,
     fontStyle,
     textDecorationLine,
+    textBounds,
   )
 }
 
+function elementContainsOnlyText(element: HTMLElement): boolean {
+  if (element.childNodes.length === 0) {
+    return false
+  }
+  for (const node of element.childNodes) {
+    const isForText =
+      node.nodeType === Node.TEXT_NODE ||
+      (node.nodeType === Node.ELEMENT_NODE && node.nodeName === 'BR')
+    if (!isForText) {
+      return false
+    }
+  }
+  return true
+}
+
+function getTextBounds(element: HTMLElement, scale: number): DOMRect | null {
```

Comment:
This feels redundant with `getCanvasRectangleFromElement` with the `withContent = 'with-content'` parameter value. 
And that is used to fill the `specialSizeMeasurements.globalFrameWithTextContent`, which is also something similar (the difference is that `globalFrameWithTextContent` is a union of the text bounds and the global frame).

Can we somehow unify how these are calculated? E.g. use `getTextBounds` in `getCanvasRectangleFromElement` too, so we don't have two ways to calculate text size.

Reply:
would something [`5e546a4` (#4202)](https://github.com/concrete-utopia/utopia/pull/4202/commits/5e546a47f8714eb570cf12f6c2d9d1816afe9b91) work?